pkg/loop/loop_linux.go

```diff
@@ -2,22 +2,79 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package loop provides an interface to interacting with Linux loop devices.
+//
+// A loop device exposes a regular file as if it were a block device.
 package loop
 
 import (
+	"github.com/u-root/u-root/pkg/mount"
 	"golang.org/x/sys/unix"
 )
 
+// Loop represents a regular file exposed as a loop block device.
+//
+// Loop implements mount.Mounter.
+type Loop struct {
+	// Dev is the loop device path.
+	Dev string
+
+	// Source is the regular file to use as a block device.
+	Source string
+
+	// FSType is the file system to use when mounting the block device.
+	FSType string
+
+	// Data is the data to pass to mount(2).
+	Data string
+
+	// Mounted indicates whether the device has been mounted.
+	Mounted bool
+
+	// dir is the directory the block device was mounted on.
+	dir string
+}
+
+// New initializes a Loop struct and allocates a loodevice to it.
+//
+// source is the file to use as a loop block device. target is the directory
+// the device should be mounted on.
+func New(source, fstype string, data string) (mount.Mounter, error) {
+	devicename, err := FindDevice()
+	if err != nil {
+		return nil, err
+	}
+	if err := SetFile(devicename, source); err != nil {
+		return nil, err
+	}
+	return &Loop{
+		Dev:    devicename,
+		Source: source,
+		FSType: fstype,
+		Data:   data,
+	}, nil
+}
+
+// Mount mounts the provided source file, with type fstype, and flags and data options
+// (which are usually 0 and ""), using the allocated loop device.
+func (l *Loop) Mount(path string, flags uintptr) error {
+	l.dir = path
+	if err := unix.Mount(l.Dev, path, l.FSType, flags, l.Data); err != nil {
+		return err
+	}
+	l.Mounted = true
```

Comment:
Isn't this a bit dicey? Some rando can umount it behind your back and then this is not going to be correct. I'm not sure why it is needed.

Reply:
This is something I didn't change from the existing loop code, just moved it - yeah, it kinda is

Reply:
file an issue and let it go in as is?